os/board/rtl8730e/src/rtl8730e_ist415.c

```diff
@@ -133,7 +133,7 @@ static void rtl8730e_ist415_power_on(struct ist415_config_s *dev)
 	struct rtl8730e_ist415_s *priv = (struct rtl8730e_ist415_s *)dev->priv;
 	GPIO_WriteBit(IST415_GPIO_RESET_PIN, PIN_HIGH);
 	DelayMs(1);  /* Wait for stable voltage before i2c commands issued */
-	priv->i2c = up_i2cinitialize(IST415_I2C_PORT);	/* Workaround: IC20 write hang issue */
+	amebasmart_i2c_init(priv->i2c);
```

Comment:
amebasmart_i2c_init takes in amebasmart_i2c_priv_s, but here we passed down i2c_dev_s, there is not casting in amebasmart_i2c_init

Reply:
Hello @zhongnuo-tang, the code is in BSP
sorry, I just raise this changes for solve quickly this issue.
I think, your changing is better about this issue . 

Could you guide kindly how to make better this? or you can raise your change.

Reply:
how about changing to below
```	
struct amebasmart_i2c_priv_s *priv_i2c = (struct amebasmart_i2c_priv_s *)priv->i2c;
amebasmart_i2c_init(priv_i2c);
```